### PR TITLE
[Merged by Bors] - use zap in p2p package

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

Part of the effort to switch from log to zap.

## Test Plan

existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: fetch/fetch_test.go
Comment (fasmat) [inline fetch/fetch_test.go]: Instead of `logtest.New` you can now use `zaptest.NewLogger` 🙂

File: fetch/p2p_test.go
Comment (fasmat) [inline fetch/p2p_test.go]: If you merge this after the other PR you can also use `zaptest.NewLogger` here because `log.Logger` isn't used any more

File: node/relay.go
Comment (fasmat) [inline node/relay.go:37]: The type of this variable is `log.Level` which is just an alias to `zapcore.Level`. Should we change that too?

File: p2p/host_test.go
Comment (fasmat) [inline p2p/host_test.go]: Should we give the individual hosts different names so we can distinguish them in the test logs?

File: p2p/ping_test.go
Comment (fasmat) [inline p2p/ping_test.go]: Same here; giving the two hosts different names would make it clearer in the testlogs which messages are from which host 🙂

File: p2p/pubsub/pubsub_test.go
Comment (fasmat) [inline p2p/pubsub/pubsub_test.go:29]: Here too, every host should probably have its own name in the logger.

File: fetch/fetch_test.go
Comment (poszu) [reply fetch/fetch_test.go]: Only after #6109 is merged :) I will update this PR once it's merged.

**Review by fasmat — APPROVED** [review]

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**
